src/include/iscsi-lio-server/TargetData.rb

```diff
@@ -0,0 +1,594 @@
+    @re_iqn_name = Regexp.new(/iqn\.\d{4}-\d{2}\.[\w\.:\-]+/)
+
+    @re_eui_target = Regexp.new(/eui\.\w+\s\.+\s\[TPGs:\s\d+\]/)
+    @re_eui_name = Regexp.new(/eui\.\w+/)
+
+    @re_tpg = Regexp.new(/tpg\d+\s/)
+
+    @re_acls_group = Regexp.new(/acls\s\.+\s\[ACLs\:\s\d+\]/)
+
+    @re_acl_iqn_rule = Regexp.new(/iqn\.\d{4}\-\d{2}\.[\w\.:\-]+\s\.+\s\[[\w\-\s\,]*Mapped\sLUNs\:\s\d+\]/)
+    @re_acl_eui_rule = Regexp.new(/eui\.\w+\s\.+\s\[[\w\-\s\,]*Mapped\sLUNs\:\s\d+\]/)
+
+    #match a line like this:
+    #mapped_lun1 .......................................................................... [lun2 fileio/iscsi_file1 (rw)]
+    @re_mapped_lun_line = Regexp.new(/mapped_lun\d+\s\.+\s\[lun\d+\s/)
+
+    # match the mapped lun like "mapped_lun1", we matched one more \s here to aovid bugs in configfs / targetcli
+    # mismatch, need to strip when use
+    @re_mapping_lun = Regexp.new(/mapped_lun\d+\s/)
+
+    #match the mapped lun, like "[lun2" in "[lun2 fileio/iscsi_file1 (rw)]", we matched one more \s to avoid bugs.
+    @re_mapped_lun = Regexp.new(/\[lun\d+\s/)
+
+    #match a line like "| | | | o- lun2 ...................... [fileio/iscsi_file1 (/home/lszhu/target1.raw) (default_tg_pt_gp)]"
+    #or "o- lun0 .................................................................. [block/iscsi_sdb (/dev/sdb) (default_tg_pt_gp)]"
+    @re_lun = Regexp.new(/\-\slun\d+\s\.+\s\[(fileio|block)\//)
+    #match lun number like lun0, lun1, lun2....
+    @re_lun_num = Regexp.new(/\-\slun\d+\s/)
+    #match lun name like [fileio/iscsi_file1 or [block/iscsi_sdb
+    @re_lun_name = Regexp.new(/\[(fileio|block)\/[\w\_\-\d]+\s/)
+    #match lun patch like:(/home/lszhu/target1.raw) or (/dev/sdb)
+    @re_lun_path = Regexp.new(/[(]\/(\w|\.|\/)+[)]/)
+    # match portal like 0.12.121.121:3260
+    @re_portal = Regexp.new(/(\d{1,3}\.){3}\d{1,3}:\d{1,5}/)
+    #iqn_name or eui_name would be a MatchData, but target_name would be a string.
+    @iqn_name= nil
+    @eui_name= nil
+    @target_name = nil
+    @initiator_name =  nil
+
+    #tgp_name would be a MatchData, but tgp_num should be a string.
+    @tpg_name = nil
+    @tpg_num = nil
+
+    #the string for a mapping lun, like mapped_lun1
+   @mapping_lun_name = nil
+    #the string for a mapped lun, like "lun2" in "[lun2 fileio/iscsi_file1 (rw)]"
+    @mapped_lun_name = nil
+
+    #will store anything match our regexp
+    @match = nil
+
+    # A pointer points to the target in the list that we are handling.
+    @current_target = nil
+    # A pointer points to the tpg in the target that we are handling.
+    @current_tpg = nil
+    # A pointer points to the acls group
+    @current_acls_group = nil
+    #A pointer points to the acl rule for a specific initiator we are handling
+    @current_acl_rule = nil
+
+    # the command need to execute  and the result
+    @cmd = nil
+    @cmd_out = nil
+    @targets_list = TargetList.new
+    self.analyze
+  end
+
+
+  def analyze()
```

Comment:
This is a 120 line long parsing method without tests. This will break, if it is not broken already. Testing parsing methods is usually easy.